### ISSUE TITLE
Getting different forecast in Qlik Sense with Prophet

### ISSUE THREAD

**Issue by ijlorant** [nabeel-oz/qlik-py-tools#78]
I'm trying to reproduce in Qlik Sense the forecasting example found in the Prophet docs where a multivariate regressor it's used but I'm getting different kind of forecast with less performance than applying the model direct in Python without the SSE funcionality.

**To Reproduce**
1. Using bike rentals, holidays and weather conditions (temp, rain, sun, wind) datasets from this example: https://nbviewer.jupyter.org/github/nicolasfauchereau/Auckland_Cycling/blob/master/notebooks/Auckland_cycling_and_weather.ipynb
2. Using the same parameters to the model 
3. Using the same period of time to train and test sets

**Expected behavior**
Expecting to have the same forecasting behaivor since we are using the same parameters and random seed for the model

**Screenshots**
1.- Datasets
- Python
![image](https://user-images.githubusercontent.com/8691464/70875538-e6181d80-1f7b-11ea-86d1-5965b31c7289.png)

- Pytools
![image](https://user-images.githubusercontent.com/8691464/70875556-f3cda300-1f7b-11ea-87e0-7d040feae72a.png)

2.- Parameters
- Python
![image](https://user-images.githubusercontent.com/8691464/70875587-0647dc80-1f7c-11ea-9661-f0bb8f86b8ea.png)

- Pytools
![image](https://user-images.githubusercontent.com/8691464/70875595-0b0c9080-1f7c-11ea-875c-1218137e380b.png)

3.- Same periods - Different Results
- Python
![image](https://user-images.githubusercontent.com/8691464/70875616-22e41480-1f7c-11ea-9e04-f5c660e6262e.png)

- Pytools
![image](https://user-images.githubusercontent.com/8691464/70875622-27103200-1f7c-11ea-8a58-eccb6fcedf84.png)

**Environment:**
 - OS: Windows 10
 - Qlik Sense Desktop
 - SSE Version: qlik-py-tools version 6.2

I've tried but can't found any possible source of the difference in both.

**Comment by nabeel-oz** [comment]
Hi @ijlorant ,

One difference I can spot is the `mcmc_samples` parameter which is set to `300` in your notebook but not specified when using PyTools which would default it to `0`.

I'd also set the `debug=true` argument in the PyTools function and check the log in the terminal to confirm that the additional regressors are being interpreted correctly.

Also, when using the line chart with a continuous axis in Qlik Sense, some data points can be smoothed out until the user zooms in. So I would use a table and compare the exact values to be sure they match.

**Comment by ijlorant** [comment]
Hi @nabeel-oz 

I've noticed that I didn't include the mcmc_samples parameters in my SS but indeed I did try with that parameter too and got almost the same result.

I think that it's not about the line chart because I've looked the forecasting values in a table and everything looks ok but looking to the log file it seems that the extra regressors are not included to fit the model.

- Input DF
![image](https://user-images.githubusercontent.com/8691464/70926136-2d40f580-1ff2-11ea-9de4-57392bf5bd6a.png)

- Future DF
![image](https://user-images.githubusercontent.com/8691464/70926176-3f229880-1ff2-11ea-8003-be2152eb20ac.png)

Let me look if I can find why and in any case, I'll let you know what happened with the issue.

**Comment by nabeel-oz** [comment]
Yes, that confirms that the additional regressors are not getting passed correctly. Try plotting your `vFeatures` variable against the `FORECAST_DATE` dimension to see if that is looking ok.

The SSE will fall back to a basic model ignoring the additional regressors if it only finds one unique value (most likely NULL) in the column.

This is done at line 663 in _prophet.py:
```
        # Check if the regressors column is empty
        if len(self.regressors_df.regressors.unique()) == 1:
            # Return without further processing
            self.has_regressors = False
            return None
```

**Comment by ijlorant** [comment]
Hi @nabeel-oz 

I've found the problem, it was a wrong data relation in QS between the extra regressors and the forecasting calendar. Now PyTool reproduce the same forecast than the notebook.

![image](https://user-images.githubusercontent.com/8691464/71010416-65f3d400-20b1-11ea-9ec4-338ddd6b2e9d.png)

Maybe I'll ask for your help soon in another case related to Keras but for now thanks for your help.

**Comment by nabeel-oz** [comment]
Great, glad you've got it working.
